care/facility/tests/test_external_result_api.py

```diff
@@ -0,0 +1,177 @@
+        self.assertEqual(response.data.get("name"), patient_external_test.name)
+
+    def test_update_patient_external_result(self):
+        sample_data = {
+            "address": "Upload test address Updated",
+        }
+        response = self.client.put(
+            f"/api/v1/external_result/{self.external_result.id}/", sample_data
+        )
+        self.external_result.refresh_from_db()
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(self.external_result.address, sample_data["address"])
+
+    def test_update_patch_patient_external_result(self):
+        sample_data = {
+            "address": "Upload test address Updated patch",
+        }
+        response = self.client.patch(
+            f"/api/v1/external_result/{self.external_result.id}/", sample_data
+        )
+        self.external_result.refresh_from_db()
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(self.external_result.address, sample_data["address"])
+
+    def test_delete_patient_external_result(self):
+        response = self.client.delete(
+            f"/api/v1/external_result/{self.external_result.id}/"
+        )
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertFalse(
+            PatientExternalTest.objects.filter(id=self.external_result.id).exists()
+        )
+
+    def test_no_data_upload(self):
+        response = self.client.post(
+            "/api/v1/external_result/bulk_upsert/", sample_data={}
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response.data["sample_tests"], "No Data was provided")
+
+    def test_different_district_upload(self):
+        sample_data = {
+            "sample_tests": [
+                {
+                    "district": "Random_district",
+                    "srf_id": "00/EKM/0000",
+                    "name": "Test Upload0",
+                    "age": 24,
+                    "age_in": "years",
+                    "gender": "m",
+                    "mobile_number": 8888888888,
+                    "address": "Upload test address",
+                    "ward": self.ward.number,
+                    "local_body": str(self.local_body.name),
+                    "local_body_type": "municipality",
+                    "source": "Secondary contact aparna",
+                    "sample_collection_date": "2020-10-14",
+                    "result_date": "2020-10-14",
+                    "test_type": "Antigen",
+                    "lab_name": "Karothukuzhi Laboratory",
+                    "sample_type": "Ag-SD_Biosensor_Standard_Q_COVID-19_Ag_detection_kit",
+                    "patient_status": "Asymptomatic",
+                    "is_repeat": "NO",
+                    "patient_category": "Cat 17: All individuals who wish to get themselves tested",
+                    "result": "Negative",
```

Comment:
now that you have a utility function you can replace this json with get_patient_external_test_data(...).update(...)